Double slash in federated cache node target_url

# Bug Report

## Describe the Bug

_Federated catalog inserts the target_url with one slash at the end of the URL generating a double slash when concat the rest of the URL path on crawler execution._

### Expected Behavior

_The last slash needs to be removed from the end of the base URL._

### Observed Behavior

_The generated URL is like http://IP:PORT//api/v1/ids/data_
_Need to be http://IP:PORT/api/v1/ids/data_

## Steps to Reproduce

Steps to reproduce the behavior:
1. Start the truzztport setup
2. Take a look if you do not have a console error 
3. Go to the database and take a look if the URL saved does not have the last slash.

## Detailed Description

SEVERE 2023-08-08T10:50:44.743783211 org.eclipse.edc.spi.EdcException: Received an error from connector (http://192.168.1.113:13011//api/v1/ids/data): 400 Bad Request


SEVERE 2023-08-08T11:51:05.567549997 ExecutionManager: Unexpected exception happened during in crawler Crawler-73d1ee25-d7e1-4970-be69-0368a3401638
java.util.concurrent.CompletionException: org.eclipse.edc.spi.EdcException: Received an error from connector (http://192.168.0.4:13011//api/v1/ids/data): 400 Bad Request
        at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:315)
        at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:320)
        at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:649)
        at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
        at java.base/java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2147)
        at dev.failsafe.spi.FailsafeFuture.completeResult(FailsafeFuture.java:97)
        at dev.failsafe.AsyncExecutionImpl.complete(AsyncExecutionImpl.java:159)
        at java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:863)
        at java.base/java.util.concurrent.CompletableFuture.uniWhenCompleteStage(CompletableFuture.java:887)
        at java.base/java.util.concurrent.CompletableFuture.whenComplete(CompletableFuture.java:2325)
        at dev.failsafe.AsyncExecutionImpl.executeAsync(AsyncExecutionImpl.java:149)
        at dev.failsafe.AsyncExecutionImpl.record(AsyncExecutionImpl.java:100)
        at dev.failsafe.AsyncExecutionImpl.recordResult(AsyncExecutionImpl.java:106)
        at dev.failsafe.okhttp.FailsafeCall$1.onResponse(FailsafeCall.java:145)
        at okhttp3.internal.connection.RealCall$AsyncCall.run(RealCall.kt:519)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
        at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: org.eclipse.edc.spi.EdcException: Received an error from connector (http://192.168.0.4:13011//api/v1/ids/data): 400 Bad Request
        at org.eclipse.edc.protocol.ids.api.multipart.dispatcher.sender.IdsMultipartSender.lambda$send$0(IdsMultipartSender.java:174)
        at org.eclipse.edc.connector.core.base.EdcHttpClientImpl.lambda$executeAsync$1(EdcHttpClientImpl.java:81)
        at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:646)

--- a/core/federated-catalog/src/main/java/org/eclipse/edc/catalog/cache/query/IdsMultipartNodeQueryAdapter.java
+++ b/core/federated-catalog/src/main/java/org/eclipse/edc/catalog/cache/query/IdsMultipartNodeQueryAdapter.java
@@ -53,7 +53,12 @@ public class IdsMultipartNodeQueryAdapter implements NodeQueryAdapter {
     private String getNodeUrl(UpdateRequest updateRequest) {
         var url = updateRequest.getNodeUrl();
         if (!url.endsWith("/ids/data")) {
-            url += "/api/v1/ids/data";
+            // prevents double slash
+            if (url.endsWith("/")) {
+                url += "api/v1/ids/data";
+            } else {
+                url += "/api/v1/ids/data";
+            }
         }
 
         return url;

